relay-server/src/actors/global_config.rs

```diff
@@ -166,6 +167,39 @@ impl Status {
     }
 }
 
+#[derive(Clone)]
+pub struct GlobalConfigHandle {
+    watch: watch::Receiver<Status>,
+}
+
+impl GlobalConfigHandle {
+    /// Creates a new global config handle with a fixed global config.
+    #[cfg(test)]
+    pub fn fixed(config: GlobalConfig) -> Self {
+        let (_, watch) = watch::channel(Status::Ready(Arc::new(config)));
+        Self { watch }
+    }
+
+    /// Returns the currently loaded or a default global config.
+    ///
+    /// When no global config has been received from upstream yet,
+    /// this will return a default global config.
+    pub fn current(&self) -> Arc<GlobalConfig> {
+        match &*self.watch.borrow() {
+            Status::Ready(config) => Arc::clone(config),
+            Status::Pending => Default::default(),
+        }
```

Comment:
IIUC, the purpose of `Pending` is to indicate there's no global config loaded yet, but this accessor is returning one. What do you think if we return `Option<Arc<GlobalConfig>>` instead, and return `None` if the status is pending? (or a similar approach)

Reply:
The problem is, how do we actually want to deal with that case? If all we do is `.unwrap_or_default()` then I prefer this.

It's a bit unfortunate that this is so much implicit behaviour, currently the only time we use this (in the processor) this should never be the case since envelopes are delayed until a global config is loaded.

Of course we can always fallback to this pattern:
```
let Some(global_config) = self.global_config.current() else { relay_log::error!(...); return; }
```
But that does make things a lot more annoying to deal with, e.g. what would we return from `light_normalize_event` , an error? Or do we need to go back to passing the global config around everywhere (which I kind of wanted to get rid of).

Reply:
Hmm, that makes sense. To be honest, I don't like any alternative I can think of nor do I have strong opinions.